cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/coprocessor/hbase11/DefaultTransactionProcessor.java

```diff
@@ -83,55 +87,91 @@ public void stop(CoprocessorEnvironment e) throws IOException {
   }
 
   @Override
+  public void postFlush(ObserverContext<RegionCoprocessorEnvironment> e) throws IOException {
+    reloadPruneState(e.getEnvironment());
+    super.postFlush(e);
+  }
+
+  @Override
+  public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEnvironment> c, Store store,
+                                               List<? extends KeyValueScanner> scanners, ScanType scanType,
+                                               long earliestPutTs, InternalScanner s,
+                                               CompactionRequest request) throws IOException {
+    reloadPruneState(c.getEnvironment());
+    return super.preCompactScannerOpen(c, store, scanners, scanType, earliestPutTs, s, request);
+  }
+
+  @Override
   protected void ensureValidTxLifetime(RegionCoprocessorEnvironment env, OperationWithAttributes op,
                                        @Nullable Transaction tx) throws IOException {
     if (tx == null) {
       return;
     }
 
-    long maxLifetimeMillis;
-    if (txMaxLifetimeMillis == null) {
-      Configuration conf = getConfiguration(env);
-      if (conf != null) {
-        this.txMaxLifetimeMillis = TimeUnit.SECONDS.toMillis(conf.getInt(TxConstants.Manager.CFG_TX_MAX_LIFETIME,
-                                                                         TxConstants.Manager.DEFAULT_TX_MAX_LIFETIME));
-        maxLifetimeMillis = this.txMaxLifetimeMillis;
-      } else {
-        // Get maxLifetimeMillis from transaction attributes
+    long currMaxLifetimeinMillis = getCurrMaxLifetime(env, op, cConfCache.getTxMaxLifetimeMillis());
+    boolean validLifetime =
```

Comment:
We can't since we don't set that variable if we get that value from the client. Should I change the logic so that we set txMaxLifetimeMillis in that case as well?